pkg/providers/instancetype/instancetype.go

```diff
@@ -165,30 +159,73 @@ func (p *DefaultProvider) List(ctx context.Context, nodeClass *v1.EC2NodeClass)
 	), nil
 }
 
-func (p *DefaultProvider) resolveInstanceTypes(
-	ctx context.Context,
-	nodeClass *v1.EC2NodeClass,
-	amiHash uint64,
-) []*cloudprovider.InstanceType {
-	zonesToZoneIDs := lo.SliceToMap(nodeClass.Status.Subnets, func(s v1.Subnet) (string, string) {
-		return s.Zone, s.ZoneID
-	})
-	return lo.FilterMap(p.instanceTypesInfo, func(info ec2types.InstanceTypeInfo, _ int) (*cloudprovider.InstanceType, bool) {
-		it := p.instanceTypesResolver.Resolve(ctx, info, p.instanceTypesOfferings[string(info.InstanceType)].UnsortedList(), zonesToZoneIDs, nodeClass)
-		if it == nil {
-			return nil, false
+func (p *DefaultProvider) Get(ctx context.Context, nodeClass *v1.EC2NodeClass, name ec2types.InstanceType) (*cloudprovider.InstanceType, error) {
+	p.muInstanceTypesInfo.RLock()
+	p.muInstanceTypesOfferings.RLock()
+	defer p.muInstanceTypesInfo.RUnlock()
+	defer p.muInstanceTypesOfferings.RUnlock()
+
+	if len(p.instanceTypesInfo) == 0 {
+		return nil, fmt.Errorf("no instance types found")
+	}
+	if len(p.instanceTypesOfferings) == 0 {
+		return nil, fmt.Errorf("no instance types offerings found")
+	}
+	if len(nodeClass.Status.Subnets) == 0 {
+		return nil, fmt.Errorf("no subnets found")
+	}
+	var instanceType *cloudprovider.InstanceType
+	if item, ok := p.instanceTypesCache.Get(p.cacheKey(nodeClass)); ok {
+		instanceType, ok = lo.Find(item.([]*cloudprovider.InstanceType), func(i *cloudprovider.InstanceType) bool {
+			return ec2types.InstanceType(i.Name) == name
+		})
+		if !ok {
```

Comment:
If we miss it on the cache, does it make more sense to try and regenerate it? I'm not sure what real world examples of this there would be tbf, but it seems strange to just bail on a cache miss.